### ISSUE TITLE
Rich text in data consent

### ISSUE THREAD

**Issue by bjarnef** [umbraco/Umbraco.Forms.Issues#834]
Currently the Data Consent field has a property "Accept Copy". I think the label of this maybe should be something different?
Besides that it would be great it we could enter a rich text description.

For example in a newsletter signup you may want to specify a bit more detailed text and often part of the text would be linking to privacy policy, cookie policy or terms and conditions. Often there would be two links and could be inlined with the text.

**Comment by AndyButland** [comment]
Not quite answering this issue I release, but please note that we will have a rich text field type in 10.1.  So you would be able to add this below the data consent field, to provide more context and links to the various privacy pages.

E.g. to create something like this:

Yes I give permission to process by data: [X]

By filling in the form you agree to our [terms and conditions](url) and [privacy policy](url).

**Comment by bjarnef** [comment]
@AndyButland yes, that's definitely an improvement.
Would the Data Content field include an richtext description or would we need to override Data Content field to archive this?
https://our.umbraco.com/documentation/Add-ons/umbracoforms/developer/extending/Adding-a-Type#overriding-default-providers-in-umbraco-forms

Would a label like "Accept Conditions" or "Accept Consent" be better than "Accept Copy"?

**Comment by AndyButland** [comment]
No, I'm just suggesting you could add two fields to your form - one data consent field, and one rich text field underneath it.

**Comment by bjarnef** [comment]
Yeah, depending on the layout, it could be part of the Data Content field as another field would typical add some spacing below the field.

**Comment by bjarnef** [comment]
I had a though about this again and although we can add a separate rich text field, I think it would be easier to manage for content editors.

Furthermore when working with Forms templates and Custom Apply Workflows Behavior it would be easier to deal with a single field.
https://docs.umbraco.com/umbraco-forms/developer/extending/customize-default-workflows

**Comment by bjarnef** [comment]
@AndyButland wasn't there is way to override a field type, to change configuration or add an extra richtext setting, while keeping the existing field type guid?

**Comment by AndyButland** [comment]
Can't say I've tried it, but seems like it 

> @AndyButland wasn't there is way to override a field type, to change configuration or add an extra richtext setting, while keeping the existing field type guid?

Can't say I've tried it, but seems like it should work.  Make sure to remove the original field type in your composer as well as adding your new one though, otherwise you'll end up with a duplicate ID.

**Comment by bjarnef** [comment]
It seems I get an

![image](https://github.com/umbraco/Umbraco.Forms.Issues/assets/2919859/3713d590-fd13-4f90-b3dd-fbe2f0bb1ed7)

but I didn't seen duplicates although I didn't remove it:

![image](https://github.com/umbraco/Umbraco.Forms.Issues/assets/2919859/3065e017-7f82-4b57-affb-b98b41f34546)

this didn't seem to make a difference:

```
builder.WithCollectionBuilder<FieldCollectionBuilder>()
    .Exclude<DataConsent>() // core
    .Add<DataConsentField>(); // custom
```

**Comment by bjarnef** [comment]
Nevermind, the error is because of the design view, where the class name doesn't match the `DataConsent` in Forms.
So this works instead (or rename this class to match name).

```
public class DataConsentField : Umbraco.Forms.Core.Providers.FieldTypes.DataConsent
{
    /// <summary>
    /// Gets or sets the form field's body text.
    /// </summary>
    [Setting("Rich text", Description = "Enter your formatted text", SupportsPlaceholders = true, HtmlEncodeReplacedPlaceholderValues = true, View = "richtext", DisplayOrder = 10)]
    public virtual string Html { get; set; } = string.Empty;

    public override string GetDesignView() =>
        "~/App_Plugins/UmbracoForms/backoffice/Common/FieldTypes/DataConsent.html";
}
```
